packages/theme/src/Components/Snackbar/index.tsx

```diff
@@ -1,37 +1,184 @@
-import { useRef, memo } from 'react'
-import { SnackbarProvider } from 'notistack'
+import { forwardRef, useRef, memo } from 'react'
+import { keyframes } from 'tss-react'
+import {
+    SnackbarProvider,
+    SnackbarProviderProps,
+    SnackbarKey,
+    useSnackbar,
+    VariantType,
+    SnackbarMessage,
+    SnackbarContent,
+} from 'notistack'
+import { Link, Typography } from '@material-ui/core'
 import { IconButton } from '@mui/material'
-import { makeStyles } from '../../makeStyles'
+import classnames from 'classnames'
+import LaunchIcon from '@material-ui/icons/Launch'
 import { Close as CloseIcon } from '@mui/icons-material'
+import WarningIcon from '@material-ui/icons/Warning'
+import InfoIcon from '@material-ui/icons/Info'
+import DoneIcon from '@material-ui/icons/Done'
+import { LoadingIcon, RiskIcon } from '@masknet/icons'
+import { makeStyles } from '../../makeStyles'
 import { MaskColorVar } from '../../constants'
 
 export { SnackbarProvider, useSnackbar } from 'notistack'
-export type { VariantType, OptionsObject } from 'notistack'
+export type { VariantType, OptionsObject, SnackbarKey } from 'notistack'
+
+const useStyles = makeStyles()((theme) => {
+    const spinningAnimationKeyFrames = keyframes`
+to {
+  transform: rotate(360deg)
+}`
+    return {
+        root: {
+            zIndex: 9999,
+            color: MaskColorVar.textLight,
+            pointerEvents: 'inherit',
+        },
+        success: {
+            background: MaskColorVar.greenMain,
+            color: MaskColorVar.lightestBackground,
+            '& $title': {
+                color: 'inherit',
+            },
+            '& $message': {
+                color: 'inherit',
+            },
+        },
+        error: {
+            background: MaskColorVar.redMain,
+            color: MaskColorVar.lightestBackground,
+            '& $title': {
+                color: 'inherit',
+            },
+            '& $message': {
```

Comment:
https://github.com/garronej/tss-react/#composition-and-nested-selectors--syntax-

Reply:
TBH, I think `css-in-js` is painful. Hard to read and write. 😂 I personally prefer `css-modules`. Could we have a try at it?

Reply:
We need to take control of when the styles are injected therefore we can make sure they work in content scripts. If you think nested selectors are not easy to use in `makeStyles`, you can try this:

```ts
const Type1 = styled("div")`
    some css
`
const Type2 = styled("div")`
    ${Type1} & {
        some other styles.
    }
`
```